pkg/tsdb/cloudwatch/cloudwatch.go

```diff
@@ -74,16 +76,98 @@ type cloudWatchExecutor struct {
 
 func (e *cloudWatchExecutor) newSession(region string) (*session.Session, error) {
 	dsInfo := e.getDSInfo(region)
-	creds, err := getCredentials(dsInfo)
+
+	cacheKey := fmt.Sprintf("%d:%s:%s:%s:%s:%s", dsInfo.AuthType, dsInfo.AccessKey, dsInfo.Profile, dsInfo.AssumeRoleARN,
```

Comment:
Not important but I would suggest using a different separator for the cacheKey since `AssumeRoleARN` can contain `:` in order to reduce ambiguity.

Reply:
Good point @papagian

Reply:
@papagian I fixed it by escaping `:` in the key components. Please have a look.